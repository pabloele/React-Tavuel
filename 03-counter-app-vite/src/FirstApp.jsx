import PropTypes from "prop-types";

function FirstApp({ title, subtitle }) {
  if (!title) throw new Error("El title no existe");
  return (
    <>
      <h1>{title}</h1>
      <h2>{subtitle + 1}</h2>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  title: "No hay título",
  subtitle: "no hay subtítulo",
  name: "Pablo",
};

export default FirstApp;
