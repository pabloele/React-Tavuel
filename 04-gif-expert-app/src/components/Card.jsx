import PropTypes from "prop-types";

export const Card = ({ title, uri }) => {
  return (
    <div className="card">
      <img src={uri} alt={title} />
      <p>{title}</p>
    </div>
  );
};

Card.propTypes = {
  title: PropTypes.string.isReq,
  uri: PropTypes.string.isReq,
};

// export default Card;
