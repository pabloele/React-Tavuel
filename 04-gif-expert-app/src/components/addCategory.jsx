import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
  const [value, setValue] = useState("");
  const handleChange = (e) => {
    setValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    const newValue = value.trim();
    if (newValue.length <= 1) return;
    setValue("");
    onNewCategory(newValue);
  };
  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        value={value}
        placeholder="Buscar Gifs"
        onChange={handleChange}
      />
      <button>Agregar</button>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};

// export default AddCategory;
