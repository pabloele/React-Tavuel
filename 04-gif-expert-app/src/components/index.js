export * from "./addCategory";
export * from "./Card";
export * from "./GifGrid";
