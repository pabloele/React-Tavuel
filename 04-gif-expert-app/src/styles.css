/* html,
body {
  background-color: black;
  color: white;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  margin-right: 5px;
} */

* {
  background-color: gray;
  font-family: Arial, Helvetica, sans-serif;
  color: whitesmoke;
  padding: 10px;
}

body {
  padding: 20px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  /* padding: 10px 15px; */
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

button {
  padding: 1rem;
  font-size: 1rem;
  margin-right: 5px;
  background-color: brown;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px black;
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  overflow: hidden;
}
