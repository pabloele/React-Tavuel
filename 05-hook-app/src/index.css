body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

p {
  font-size: x-large;
}

a {
  margin-right: 2rem;
}
