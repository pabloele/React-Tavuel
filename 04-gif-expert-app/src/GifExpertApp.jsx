import { useState } from "react";

import { AddCategory, GifGrid } from "./components/index";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onNewCategory = (value) => {
    if (categories.includes(value)) return;
    setCategories((c) => [...c, value]);
  };
  return (
    <>
      <h1>Gif Expert App</h1>

      <AddCategory onNewCategory={onNewCategory} />

      {categories.map((c) => (
        <GifGrid category={c} key={c} />
      ))}
    </>
  );
};

export default GifExpertApp;
