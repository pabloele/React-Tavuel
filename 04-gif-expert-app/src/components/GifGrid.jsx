import useFetchGifs from "../hooks/useFetchGifs";
import { Card } from "./Card";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {isLoading ? (
          <h1>cargando...</h1>
        ) : (
          images.map((image) => <Card key={image.id} {...image} />)
        )}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
