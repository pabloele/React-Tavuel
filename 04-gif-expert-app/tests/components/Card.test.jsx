import { render, screen } from "@testing-library/react";
import { Card } from "../../src/components";

describe("Pruebas en <Card/>", () => {
  const title = "Sample title";
  const uri =
    "https://media2.giphy.com/media/Ur3robVPYVLkk/giphy.gif?cid=d7999a31gua4s7aj94uzughixjxjfcg7813thsro5bxy86dh&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<Card title={title} uri={uri} />);
    expect(container).toMatchSnapshot();
  });
});
