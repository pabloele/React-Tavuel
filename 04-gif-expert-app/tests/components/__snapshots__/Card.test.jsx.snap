// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Card/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sample title"
      src="https://media2.giphy.com/media/Ur3robVPYVLkk/giphy.gif?cid=d7999a31gua4s7aj94uzughixjxjfcg7813thsro5bxy86dh&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Sample title
    </p>
  </div>
</div>
`;
